episodes/anonymization.md

```diff
@@ -32,19 +33,352 @@ Metadata elements in imaging, such as patient names and addresses, are often cle
+
+```python
+
+from skimage.filters import gaussian
+from skimage.color import rgb2gray
+
+image_base = io.imread(image)
+image_base = rgb2gray(image_base)
+sub_im = image_base[0:78,:].copy()
+blur_sub_im = gaussian(sub_im, sigma=9)
+final_image = np.zeros(image_base.shape)
+final_image[0:78,:] = blur_sub_im
+final_image[79:,:]= image_base[79:, :]
+io.imshow(final_image)
+
+```
+
+
+```output
+```
+
+![Image after blurring in one area.](fig/blurred_us.png){alt='Non-Identifiable blurred ultrasound'}
+We could have also just make a simple zero-mask: 
+
+```python
+image_masked = io.imread(image)
+image_masked[0:78,:] = 0
+io.imshow(image_masked)
+
+```
+
+
+```output
+```
+
+![Image after masking in one area.](fig/masked_us.png){alt='Non-Identifiable masked ultrasound'}
+
+Note there are other valid solutions, but these two are very common and straightforward. 
+
+Finally, you could always just chop off the offending part so to speak, and resize:
+
+```python
+from skimage.transform import resize
+
+final_image= image_base[79:, :]
+final_image = resize(final_image,image_base.shape)
+io.imshow(final_image)
+```
+
+```output
+```
+
+![Image after crop and resize.](fig/resize_us1.png){alt='Non-Identifiable cropped ultrasound'}
+:::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+You have now seen three approaches to removing some of the visual data on a 2-D image. The same approaches can be taken on a 3D image. 
+
+Let's take another look at an image we used earlier:
+
+```python
+import matplotlib.pyplot as plt
+import SimpleITK as sitk
+from skimage import io
+from ipywidgets import interact, fixed
+from IPython.display import clear_output
+import os
+
+
+sag_image =  sitk.ReadImage("data/sitk/A1_grayT1.nrrd", sitk.sitkFloat32)
+cor_image = sitk.PermuteAxes(sag_image, [2, 1, 0])
+
+# General display function for any two 3D images
+def display_images(image1_z, image2_z, image1_npa, image2_npa, title1="Image 1", title2="Image 2"):
+    plt.subplots(1, 2, figsize=(10, 8))
+    
+    # Display the first image
+    plt.subplot(1, 2, 1)
+    plt.imshow(image1_npa[image1_z, :, :], cmap=plt.cm.Greys_r)
+    plt.title(title1)
+    plt.axis('off')
+    
+    # Display the second image
+    plt.subplot(1, 2, 2)
+    plt.imshow(image2_npa[image2_z, :, :], cmap=plt.cm.Greys_r)
+    plt.title(title2)
+    plt.axis('off')
+    
+    plt.show()
+
+# Display the sagittal and coronal views of the original image
+interact(
+    display_images,
+    image1_z=(0, sag_image.GetSize()[2] - 1),
+    image2_z=(0, cor_image.GetSize()[2] - 1),
+    image1_npa=fixed(sitk.GetArrayViewFromImage(sag_image)),
+    image2_npa=fixed(sitk.GetArrayViewFromImage(cor_image)),
+    title1=fixed("Sagittal Cut"),
+    title2=fixed("Coronal Cut")
+)
+```
+```output
+```
+![Images of SITK head.](fig/headcutsscroll.png){alt='Non-Identifiable head'}
```

Comment:
I think this image still needs to be regenerated, it doesn't quite match the code.

Reply:
I ran this by copy-pasting out of the episode into a notebook, and it works. Also, Giulia has very similar code that works this way, so I trust it.